src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs

```diff
@@ -13,245 +13,462 @@ public enum SyntaxKind : ushort
         List = GreenNode.ListKind,
 
         // punctuation
+        /// <summary>Represents <c>~</c> token.</summary>
         TildeToken = 8193,
+        /// <summary>Represents <c>!</c> token.</summary>
         ExclamationToken = 8194,
+        /// <summary>Represents <c>$</c> token.</summary>
+        /// <remarks>This is a debugger special punctuation and not related to string interpolation.</remarks>
```

Comment:
i'm not really comfortable with comments like this as i'm not sure if they are actually true or would remain true in the future.  For example if i write `return $;` will i get a DollarToken in the tree? 

Basically, i expect all tokens to be always possible.

Reply:
@CyrusNajmabadi I don't see any references to `DollarToken` in the lexer, or anywhere else where it could be created by Roslyn. Specifically, your return `return $;` example uses `SyntaxKind.BadToken`, not `DollarToken`.

Of course I can't say what will happen in the future, but I think the documentation should describe the current state. If that changes in the future, the documentation should change then.

Reply:
@svick I agree with you.

Reply:
I think it should just say it represents a `$` token. 

It can have a remark that a $ in an interested string is an interpolatedstringstarttoken.  But we should not be docing impl details in this public model API. 

If we doc Parse or ParseTokens, this sort of information could be there (but if still recommend being vaguer on this)

Reply:
@CyrusNajmabadi As a documentation reader, it would be important to me that I just don't need to do anything with the DollarToken.

Reply:
I'm not really comfortable stating that though. 1. I don't actually know if it's true. 2. Once we doc that, that's the behavior it needs to have.

Reply:
- It seems true for two reasons:

   1. No references to SyntaxKind.DollarToken except the one stated by @svick
   2. Try to do anything to get this DollarToken in a syntax tree, it simply won't happen. (not string reasoning, but just to support the first reasoning)

- If that changed in the future, I don't see any harm in updating the doc.

Reply:
I'm not comfortable with this. It feels like leaking out and making a particular promise about a current implementation detail.

Reply:
@CyrusNajmabadi The problem is: many readers may try to find out when a specific token can be useful. The wording may change to not reveal implementation detail about debugger, but IMO it should mention at least that this specific SyntaxKind is meant for internal use only or something like that. That way we don't reveal the detail about the debugger, and we also still helping documentation readers to know they don't need to play with this token, and if this changed in the future, we just update the doc.

Reply:
@CyrusNajmabadi The fact that this token is not related to string interpolation is a compat promise that will not change. The comment in its current form is better than the comment with less information.

Reply:
Note that the additional remarks were only included for tokens where it's easy to see that a reader would mistake the intent of the token based on the name or text alone.

Reply:
> Basically, i expect all tokens to be always possible.

This is not true for _valid_ syntax trees, and the documentation is not intended (or needed) to cover all manifestations of invalid syntax trees.